scio-core/src/main/scala/com/spotify/scio/util/ScioUtil.scala

```diff
@@ -82,5 +85,45 @@ private[scio] object ScioUtil {
       s"Could not get dataflow metrics of ${getMetrics.getJobId} in ${getMetrics.getProjectId}")
   }
 
+  def tempLocation(options: PipelineOptions): String = {
+    val location = Try(options.as(classOf[GcpOptions]))
+      .map(_.getGcpTempLocation).getOrElse(options.getTempLocation)
+    if (location == null) {
+      sys.props("java.io.tmpdir")
+    } else {
```

Comment:
Is this `else` needed here? Isn't there already `getOrElse(options.getTempLocation)`?